src/ssl.c

```diff
@@ -42163,51 +42883,25 @@ void* wolfSSL_GetDhAgreeCtx(WOLFSSL* ssl)
      */
     int wolfSSL_OBJ_ln2nid(const char *ln)
     {
-        static const struct {
-            const char *ln;
-            int  nid;
-        } ln2nid[] = {
-            {WOLFSSL_LN_COMMON_NAME, NID_commonName},
-            {WOLFSSL_LN_COUNTRY_NAME, NID_countryName},
-            {WOLFSSL_LN_LOCALITY_NAME, NID_localityName},
-            {WOLFSSL_LN_STATE_NAME, NID_stateOrProvinceName},
-            {WOLFSSL_LN_ORG_NAME, NID_organizationName},
-            {WOLFSSL_LN_ORGUNIT_NAME, NID_organizationalUnitName},
-            {WOLFSSL_EMAIL_ADDR, NID_emailAddress},
-            {NULL, -1}};
-
-        int i;
+        const WOLFSSL_ObjectInfo *obj_info = wolfssl_object_info;
+        size_t i, lnlen;
         WOLFSSL_ENTER("wolfSSL_OBJ_ln2nid");
-
-        if (ln == NULL) return BAD_FUNC_ARG;
-
-        /* Return certificate info ln */
-        for(i=0; ln2nid[i].ln != NULL; i++) {
-            if(XSTRNCMP(ln, ln2nid[i].ln, XSTRLEN(ln2nid[i].ln)) == 0) {
-                return ln2nid[i].nid;
+        if (ln && (lnlen = XSTRLEN(ln)) > 0) {
+            /* Accept input like "/commonName=" */
+            if (ln[0] == '/') {
+                ln++;
+                lnlen--;
             }
-        }
-
-        /* Return ECC ln */
-        #ifdef HAVE_ECC
-        {
-            int eccEnum;
-            /* Nginx uses this OpenSSL string. */
-            #ifdef WOLFSSL_NGINX
-                if (XSTRNCMP(ln, "prime256v1", 10) == 0) ln = "SECP256R1";
-                if (XSTRNCMP(ln, "secp384r1", 10) == 0) ln = "SECP384R1";
-            #endif
-            /* find based on name and return NID */
-            for (i = 0; i < ecc_sets[i].size; i++) {
-                if (XSTRNCMP(ln, ecc_sets[i].name, ECC_MAXNAME) == 0) {
-                    eccEnum = ecc_sets[i].id;
-                    /* Convert enum value in ecc_curve_id to OpenSSL NID */
-                    return EccEnumToNID(eccEnum);
+            if (ln[lnlen-1] == '=') {
```

Comment:
that works......for the access here since ln has also been incremented by 1 , but I suggest we avoid accessing arrays with a negative index value and we could add in an error out early case instead